okhttp/src/test/java/okhttp3/EventListenerTest.java

```diff
@@ -943,56 +947,80 @@ private void responseBodyFail(Protocol expectedProtocol) throws IOException {
         "ResponseBodyEnd", "ConnectionReleased", "CallEnd");
   }
 
-  @Test public void requestBodyFailHttp1OverHttps() throws IOException {
+  @Test public void requestBodyFailHttp1OverHttps() {
     enableTlsWithTunnel(false);
     server.setProtocols(asList(Protocol.HTTP_1_1));
-    requestBodyFail();
+
+    requestBodyFail(Protocol.HTTP_1_1);
   }
 
-  @Test public void requestBodyFailHttp2OverHttps() throws IOException {
+  @Test public void requestBodyFailHttp2OverHttps() {
     enableTlsWithTunnel(false);
     server.setProtocols(asList(Protocol.HTTP_2, Protocol.HTTP_1_1));
-    requestBodyFail();
-  }
 
-  @Test public void requestBodyFailHttp() throws IOException {
-    requestBodyFail();
+    requestBodyFail(Protocol.HTTP_2);
   }
 
-  private void requestBodyFail() {
-    // Stream a 256 MiB body so the disconnect will happen before the server has read everything.
-    RequestBody requestBody = new RequestBody() {
-      @Override public MediaType contentType() {
-        return MediaType.get("text/plain");
-      }
-
-      @Override public long contentLength() {
-        return 1024 * 1024 * 256;
-      }
-
-      @Override public void writeTo(BufferedSink sink) throws IOException {
-        for (int i = 0; i < 1024; i++) {
-          sink.write(new byte[1024 * 256]);
-          sink.flush();
-        }
-      }
-    };
+  @Test public void requestBodyFailHttp() {
+    requestBodyFail(null);
+  }
 
+  private void requestBodyFail(@Nullable Protocol requiredProtocol) {
     server.enqueue(new MockResponse()
         .setSocketPolicy(SocketPolicy.DISCONNECT_DURING_REQUEST_BODY));
 
+    NonCompletingRequest request = new NonCompletingRequest();
```

Comment:
nit: It's not really a request.

```suggestion
    NonCompletingRequestBody requestBody = new NonCompletingRequestBody();
```